contracts/Fundable.sol

```diff
@@ -0,0 +1,177 @@
+pragma solidity ^0.5.0;
+
+import "./IFundable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./libraries/StringUtil.sol";
+
+
+contract Fundable is IFundable, ERC20 {
+    using StringUtil for string;
+
+    struct FundableData {
+        address orderer;
+        address walletToFund;
+        uint256 value;
+        string instructions;
+        FundStatusCode status;
+    }
+
+    address public tokenOperator;
+    // walletToFund -> authorized -> true/false
+    mapping(address => mapping(address => bool)) public fundOperators;
+
+    mapping(bytes32 => FundableData) private orderedFunds;
+
+    constructor() public {
+        tokenOperator = msg.sender;
+    }
+
+    function authorizeFundOperator(address orderer) external returns (bool) {
+        require(fundOperators[msg.sender][orderer] == false, "The operator is already authorized");
+
+        fundOperators[msg.sender][orderer] = true;
+        emit FundOperatorAuthorized(msg.sender, orderer);
+        return true;
+    }
+
+    function revokeFundOperator(address orderer) external returns (bool) {
+        require(fundOperators[msg.sender][orderer], "The operator is already not authorized");
+
+        fundOperators[msg.sender][orderer] = false;
+        emit FundOperatorRevoked(msg.sender, orderer);
+        return true;
+    }
+
+    function orderFund(
+        string calldata operationId,
+        uint256 value,
+        string calldata instructions
+    ) external returns (bool)
+    {
+        return _orderFund(
+            operationId,
+            msg.sender,
+            value,
+            instructions
+        );
+    }
+
+    function orderFundFrom(
+        string calldata operationId,
+        address walletToFund,
+        uint256 value,
+        string calldata instructions
+    ) external returns (bool)
+    {
+        require(address(0) != walletToFund, "WalletToFund address must not be zero address");
+        require(_isFundOperatorFor(msg.sender, walletToFund), "This operator is not authorized");
+        return _orderFund(
+            operationId,
+            walletToFund,
+            value,
+            instructions
+        );
+    }
+
+    function cancelFund(string calldata operationId) external returns (bool) {
+        FundableData storage fund = orderedFunds[operationId.toHash()];
+        require(fund.status == FundStatusCode.Ordered, "A fund can only be cancelled in status Ordered");
+        require(fund.walletToFund == msg.sender || fund.orderer == msg.sender, "Only the wallet who receives the fund can cancel");
+        fund.status = FundStatusCode.Cancelled;
+        emit FundCancelled(fund.orderer, operationId);
+        return true;
+    }
+
+    function processFund(string calldata operationId) external returns (bool) {
+        require(tokenOperator == msg.sender, "A fund can only be processed by the fund operator");
```

Comment:
I think we need to check if its  fundoperator. If yes, maybe a modifier can be implemented and used also in orderFrom

Reply:
In the Clearable and Payoutable we have used clearingAgent and payoutAgent. The advantage I see, is that we don't use operator for the contract owner and the operators that are set in authorizeFundOperator and can call orderFundFrom for somebody else.
So would fundAgent be okay for you as well?

Reply:
Makes sense, fundAgents and check if its one ok kind

Reply:
I've added a new role FundAgent and use this one